### PR TITLE
Feature shared bus

### PR DESCRIPTION
**Pull request by m4github**
send and receive a queue between components.

### CODE REVIEW

**Comment by m4github** [comment]
restore main file changes.

**Comment by m4github** [comment]
Add comments on the shared-bus-receive function and remove extra comments.

File: main/app_main.cpp
Comment (Behn4m) [inline main/app_main.cpp:1]: why you are updating the `main.c`? if you want to test and demo the code, you can use the https://github.com/DoneTech-IoT/InterCom-test-code repo for it.

File: components/lvgl_esp32_drivers/lvgl_tft/rm67162Lilygo.h [hunk 1]
```diff
@@ -23,7 +23,7 @@ extern "C" {
 
 #define SPI_FREQUENCY         80000000
 #define TFT_SPI_MODE          0
-#define TFT_SPI_HOST          SPI2_HOST
```
Comment (Behn4m) [inline components/lvgl_esp32_drivers/lvgl_tft/rm67162Lilygo.h:26]: what is this?

File: main/app_main.cpp
Comment (Behn4m) [inline main/app_main.cpp:1]: why are you updating the `main.c?` if you want to test and demo the code, you can use the https://github.com/DoneTech-IoT/InterCom-test-code repo for it.

File: main/ServiceManger.cpp [hunk 2]
```diff
@@ -108,7 +108,7 @@ void ServiceMangerInit()
 void ServiceMangerTask(void *pvParameter)
 {
     ServiceMangerInit();
-    char pcTaskList[TASK_LIST_BUFFER_SIZE];
```
Comment (Behn4m) [inline main/ServiceManger.cpp:111]: why this line commented?

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#define UI_EVENT_FLAG	( 1 << UI_INTERFACE_ID )
+#define UI_EVENT_FLAG	( 1 << UI_INTERFACE_ID )
+
+#define BIT_23	( 1 << 23 ) //prevent overwriting bus for send before one turn of the event loop
+#define BIT_22	( 1 << 22 ) //peek to receive switch flag 
+
+static const char *TAG = "SharedBus";
+
+static EventBits_t EventBits;
+static EventGroupHandle_t EventGroupHandleLocal;
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:14]: this is not local variable

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+esp_err_t SharedBusInit(EventGroupHandle_t *EventGroupHandle,QueueHandle_t *QueueHandle)
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:16]: no unit test? check if input arguments are not `null`

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    EventGroupHandleLocal = *EventGroupHandle;
+    EventGroupHandleLocal = xEventGroupCreate();  
+    *QueueHandle = xQueueCreate(1, sizeof(SharedBusPacket_t));     
+    return 0;         
+}
+
+esp_err_t SharedBusSend(        
+    QueueHandle_t QueueHandle,
+    SharedBusPacket_t SharedBusPacket)    
+{
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:28]: add the unit test

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+esp_err_t SharedBusSend(        
+    QueueHandle_t QueueHandle,
+    SharedBusPacket_t SharedBusPacket)    
+{    
+    EventBits = xEventGroupWaitBits(
+                    EventGroupHandleLocal,   /* The event group being tested. */
+                    BIT_23,         /* The bits within the event group to wait for. */
+                    pdTRUE,        /* BIT_23 should be cleared before returning. */
+                    pdFALSE,        /* Wait for 23th bit, either bit will do. */
+                    1);/* Wait a maximum of 100ms for either bit to be set. */
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:34]: magic number?

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    EventBits = xEventGroupSetBits(
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:35]: what happens in the case of timeout? Do you want to send the next queue anyway?

Comment (Behn4m) [reply components/Utilities/SharedBus/SharedBus.cpp:35]: Sending after a timeout should be notified in the log.

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+    xQueueSend(QueueHandle, &SharedBusPacket, 1);
+    return 0;    
+}
+
+esp_err_t SharedBusRecieve(    
+    QueueHandle_t QueueHandle, 
+    SharedBusPacket_t SharedBusPacket,
+    TaskInterfaceID_t interfaceID)
+{
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:47]: add unit test at the begining of the function.

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                    EventGroupHandleLocal,   /* The event group being tested. */
+                    BIT_22,         /* The bits within the event group to wait for. */
+                    pdFALSE,        /* BIT_22 should NOT be cleared before returning. */
+                    pdTRUE,         /* Wait for 22th bit, either bit will do. */
+                    1);/*           Wait a maximum of 100ms for either bit to be set. */
+    
+    if((EventBits & BIT_22))
+    {
+        if(xQueueReceive(QueueHandle, &SharedBusPacket, 1) == pdTRUE)
+            ESP_LOGE(TAG, "Recieved-%d", SharedBusPacket.SourceID);
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:59]: 1- put code under if in `{...}` even if it is one line of code
2- I think you should `return 0` after this error

File: components/Utilities/SharedBus/SharedBus.cpp [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        //clear xQueueReceive flag
+        EventBits = xEventGroupClearBits(
+                    EventGroupHandleLocal, 
+                    BIT_22);                                 
+        ret = false;
+    }        
+    else //default state
+    {
+        if(xQueuePeek(QueueHandle, &SharedBusPacket, 1) == pdTRUE) //TODO err-handling   
+            ESP_LOGE(TAG, "Peek-%d", SharedBusPacket.SourceID);
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.cpp:74]: same comment as above

File: components/Utilities/SharedBus/SharedBus.h [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    LOG_INTERFACE_ID = 3  
+} TaskInterfaceID_t;
+
+typedef struct {
+    uint8_t SourceID;
+    uint8_t PacketID;    
+    void *data;
+} SharedBusPacket_t;
+
+esp_err_t SharedBusInit(EventGroupHandle_t *EventGroupHandle,QueueHandle_t *QueueHandle);
```
Comment (Behn4m) [inline components/Utilities/SharedBus/SharedBus.h:22]: add a Doxygen format comment for funtions

**Review by Behn4m — CHANGES_REQUESTED** [review]

**Comment by m4github** [comment]
This PR is going to be close. I will send another PR for SharedBus feature and fix every comment you posted on this PR. @Behn4m

File: components/Utilities/SharedBus/SharedBus.cpp (hunk 3 above)
Comment (m4github) [reply components/Utilities/SharedBus/SharedBus.cpp:14]: why not! @Behn4m

**Pull request closed without merging**